kie-wb-common-react/boxed-expression-editor/src/components/Table/Table.tsx

```diff
@@ -0,0 +1,364 @@
+    setTableRows((prevTableCells) => {
+      const updatedTableCells = [...prevTableCells];
+      updatedTableCells[rowIndex][columnId] = value;
+      return updatedTableCells;
+    });
+  }, []);
+
+  const generateNextAvailableColumnName: (lastIndex: number) => string = useCallback(
+    (lastIndex) => {
+      const candidateName = `${columnPrefix}${lastIndex}`;
+      const columnWithCandidateName = _.find(tableColumns, { accessor: candidateName });
+      return columnWithCandidateName ? generateNextAvailableColumnName(lastIndex + 1) : candidateName;
+    },
+    [columnPrefix, tableColumns]
+  );
+
+  const generateNextAvailableColumn = useCallback(
+    (columns: Column[]) => {
+      return {
+        accessor: generateNextAvailableColumnName(columns.length),
+        label: generateNextAvailableColumnName(columns.length),
+        dataType: DataType.Undefined,
+      };
+    },
+    [generateNextAvailableColumnName]
+  );
+
+  const handlingOperation = useCallback(
+    (tableOperation: TableOperation) => {
+      switch (tableOperation) {
+        case TableOperation.ColumnInsertLeft:
+          setTableColumns((prevTableColumns) =>
+            insertBefore(prevTableColumns, lastSelectedColumnIndex, generateNextAvailableColumn(prevTableColumns))
+          );
+          break;
+        case TableOperation.ColumnInsertRight:
+          setTableColumns((prevTableColumns) =>
+            insertAfter(prevTableColumns, lastSelectedColumnIndex, generateNextAvailableColumn(prevTableColumns))
+          );
+          break;
+        case TableOperation.ColumnDelete:
+          setTableColumns((prevTableColumns) => deleteAt(prevTableColumns, lastSelectedColumnIndex));
+          break;
+        case TableOperation.RowInsertAbove:
+          setTableRows((prevTableRows) => insertBefore(prevTableRows, lastSelectedRowIndex, {}));
+          break;
+        case TableOperation.RowInsertBelow:
+          setTableRows((prevTableRows) => insertAfter(prevTableRows, lastSelectedRowIndex, {}));
+          break;
+        case TableOperation.RowDelete:
+          setTableRows((prevTableRows) => deleteAt(prevTableRows, lastSelectedRowIndex));
+          break;
+      }
+      setShowTableHandler(false);
+    },
+    [generateNextAvailableColumn, lastSelectedColumnIndex, lastSelectedRowIndex]
+  );
+
+  const defaultColumn = {
+    minWidth: 38,
+    width: 150,
+    maxWidth: 600,
+    Cell: useCallback((cellRef: EditableCellProps) => {
+      const column = cellRef.column as ColumnInstance;
+      return column.canResize ? EditableCell(cellRef) : cellRef.value;
+    }, []),
+  };
+
+  const getThProps = (columnIndex: number) => ({
+    onContextMenu: (e: ContextMenuEvent) => {
+      e.preventDefault();
+      setTableHandlerAllowedOperations([
+        TableOperation.ColumnInsertLeft,
+        TableOperation.ColumnInsertRight,
+        ...(tableColumns.length > 2 ? [TableOperation.ColumnDelete] : []),
```

Comment:
This would allow to delete row number column

Reply:
That is a great catch! Thanks @jomarko